parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -125,7 +134,9 @@ public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetric
         } else if (!stats.isEmpty()) {
           increment(nullValueCounts, fieldId, stats.getNumNulls());
 
-          if (metricsMode != MetricsModes.Counts.get()) {
+          // since Parquet includes NaN for upper/lower bounds of floating point column that we don't want,
+          // we have tracked metrics for such columns ourselves and thus do not need to rely on Parquet's stats.
```

Comment:
You might want to rephrase this not to assume the reason why `fieldMetricsMap` contains metrics. If there are metrics gathered by Iceberg for a column, then we should use those instead of the ones from Parquet. The NaN handling for float and double columns is one case of that. But there may be others in the future.